flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/BigQuerySourceSplitAssigner.java

```diff
@@ -0,0 +1,247 @@
+                            String dataset = result.getDestinationDataset().get();
+                            String table = result.getDestinationTable().get();
+                            LOG.info(
+                                    "After BigQuery query execution, switching connect options"
+                                            + " to read from table {}.{}.{}",
+                                    projectId,
+                                    dataset,
+                                    table);
+                            return this.readOptions
+                                    .getBigQueryConnectOptions()
+                                    .toBuilder()
+                                    .setProjectId(projectId)
+                                    .setDataset(dataset)
+                                    .setTable(table)
+                                    .build();
+                        });
+    }
+
+    private Optional<QueryResultInfo> runQuery(String query) {
+        return BigQueryServicesFactory.instance(this.readOptions.getBigQueryConnectOptions())
+                .queryClient()
+                .runQuery(this.readOptions.getQueryExecutionProject(), query);
+    }
+
+    public void open() {
+        LOG.info("BigQuery source split assigner is opening.");
+        if (!initialized) {
+            BigQueryConnectOptions connectionOptions =
+                    fetchOptionsFromQueryRun().orElse(this.readOptions.getBigQueryConnectOptions());
+            try (BigQueryServices.StorageReadClient client =
+                    BigQueryServicesFactory.instance(connectionOptions).storageRead()) {
+                String parent = String.format("projects/%s", connectionOptions.getProjectId());
+
+                String srcTable =
+                        String.format(
+                                "projects/%s/datasets/%s/tables/%s",
+                                connectionOptions.getProjectId(),
+                                connectionOptions.getDataset(),
+                                connectionOptions.getTable());
+
+                // We specify the columns to be projected by adding them to the selected fields,
+                // and set a simple filter to restrict which rows are transmitted.
+                TableReadOptions.Builder optionsBuilder = TableReadOptions.newBuilder();
+
+                readOptions
+                        .getColumnNames()
+                        .forEach(name -> optionsBuilder.addSelectedFields(name));
+                optionsBuilder.setRowRestriction(readOptions.getRowRestriction());
+
+                TableReadOptions options = optionsBuilder.build();
+
+                // Start specifying the read session we want created.
+                ReadSession.Builder sessionBuilder =
+                        ReadSession.newBuilder()
+                                .setTable(srcTable)
+                                .setDataFormat(DataFormat.AVRO)
```

Comment:
Are we planing to use AVRO as the data format for long or just the beginning? It is less performance than ARROW

Reply:
also previously discussed, see: https://github.com/GoogleCloudDataproc/flink-bigquery-connector/pull/44#discussion_r1287771847

we created https://github.com/GoogleCloudDataproc/flink-bigquery-connector/issues/57 to address this.